src/methods.jl

```diff
@@ -99,6 +99,7 @@ function iterate(s::Ptr{S}, state) where S<:VectorSxp
     (s[state], state)
 end
 
+Base.eachindex(s::Ptr{<:VectorSxp}) = Base.OneTo(length(s))
```

Comment:
Feels weird to define `eachindex` for a `Ptr` but we also have `iterate` and `getindex`...

Reply:
yeah, it does, but then again we're treating R native arrays as iterators without copying...